src/app/pages/questions/services/audio-record.service.ts

```diff
@@ -8,41 +8,33 @@ declare var Media: any // stops errors w/ cordova-plugin-media-with-compression
 
 @Injectable()
 export class AudioRecordService {
-  isRecording: boolean = false
+  isRecording: boolean
   audio
   fileName = 'audio.m4a'
-  recordingTimeout
 
-  constructor(private file: File, private device: Device) {
-    // NOTE: Kill recording on load
-    this.stopAudioRecording()
-  }
+  constructor(private file: File, private device: Device) {}
 
-  startAudioRecording(length): Promise<any> {
+  startAudioRecording(): Promise<any> {
     return new Promise((resolve, reject) => {
       if (!this.isRecording) this.isRecording = true
       else reject()
 
       this.audio = new Media(this.getFilePath(), this.success, this.failure)
       if (this.isRecording) {
-        this.audio.startRecordWithCompression(DefaultAudioRecordOptions)
-        this.recordingTimeout = setTimeout(() => {
-          console.log('Time up for recording')
-          this.stopAudioRecording()
-          resolve(this.readAudioFile())
-        }, length)
+        return this.audio.startRecordWithCompression(DefaultAudioRecordOptions)
       } else {
         reject()
       }
     })
   }
 
-  stopAudioRecording() {
-    if (this.isRecording) {
-      this.audio.stopRecord()
-      this.isRecording = false
-      clearTimeout(this.recordingTimeout)
-    }
+  stopAudioRecording(): Promise<any> {
+    this.audio.stopRecord()
+    this.isRecording = false
+    return this.readAudioFile().then(data => {
+      this.destroy()
+      return data
+    })
```

Comment:
Maybe we can name it something like stopAndGetAudioRecording() ?

Reply:
Ok I'll fix it.